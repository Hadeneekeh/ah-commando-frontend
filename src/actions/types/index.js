export const SET_CURRENT_USER = 'SET_CURRENT_USER';
export const MODAL_OPEN = 'MODAL_OPEN';
export const MODAL_CLOSE = 'MODAL_CLOSE';
export const LOADING = 'LOADING';
export const NOT_LOADING = 'NOT_LOADING';
export const GET_PROFILE_START = 'GET_PROFILE_START';
export const GET_PROFILE_SUCCESS = 'GET_PROFILE_SUCCESS';
export const GET_PROFILE_FAILURE = 'GET_PROFILE_FAILURE';

export const GET_ARTICLE_START = 'GET_ARTICLE_START';
export const GET_ARTICLE_SUCCESS = 'GET_ARTICLE_SUCCESS';
export const GET_ARTICLE_FAILURE = 'GET_ARTICLE_FAILURE';

export const EDIT_PROFILE_START = 'EDIT_PROFILE_START';
export const EDIT_PROFILE_SUCCESS = 'EDIT_PROFILE_SUCCESS';
export const EDIT_PROFILE_FAILURE = 'EDIT_PROFILE_FAILURE';

export const POST_IMAGE_START = 'POST_IMAGE_START';
export const POST_IMAGE_SUCCESS = 'POST_IMAGE_SUCCESS';
export const POST_IMAGE_FAILURE = 'POST_IMAGE_FAILURE';

export const UPDATE_FILTER = 'UPDATE_FILTER';
export const DISPLAY_FILTER = 'DISPLAY_FILTER';
export const REMOVE_FILTER = 'REMOVE_FILTER';
export const UPDATE_ARTICLES = 'UPDATE_ARTICLES';
export const UPDATE_SEARCHQUERY = 'UPDATE_SEARCHQUERY';
export const UPDATE_PAGENUMBER = 'UPDATE_PAGENUMBER';
export const CREATE_ARTICLE = 'CREATE_ARTICLE';
export const READ_ARTICLE = 'READ_ARTICLE';

export const UPDATE_CATEGORY_ARTICLES = 'UPDATE_CATEGORY_ARTICLES';
export const UPDATE_CATEGORY = 'UPDATE_CATEGORY';
export const UPDATE_MENU_ITEM = 'UPDATE_MENU_ITEM';
export const UPDATE_CATEGORY_PAGENUMBER = 'UPDATE_CATEGORY_PAGENUMBER';
