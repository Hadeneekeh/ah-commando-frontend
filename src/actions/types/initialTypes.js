export const IS_REDUCING = 'IS_REDUCING';
export const HAS_REDUCED = 'HAS_REDUCED';
