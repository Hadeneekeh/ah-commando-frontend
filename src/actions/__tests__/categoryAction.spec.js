/* eslint-disable no-unused-vars */
import configureMockStore from 'redux-mock-store';
import * as types from '@Actions/types';
import thunk from 'redux-thunk';
import nock from 'nock';
import {
  updateArticles,
  updateCategory,
  updateMenuItem,
  getCategoryArticles,
} from '@Actions/categoryActions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const url = 'https://a-haven-staging.herokuapp.com/api/v1';

describe('actions', () => {
  it('should update articles', () => {
    const payload = {
      category: 'technology',
      Articles: [{}],
    };

    const expectedAction = {
      type: types.UPDATE_CATEGORY_ARTICLES,
      payload,
    };
    expect(updateArticles(payload)).toEqual(expectedAction);
  });

  it('should update the menu item', () => {
    const payload = 1;

    const expectedAction = {
      type: types.UPDATE_MENU_ITEM,
      payload,
    };

    expect(updateMenuItem(payload)).toEqual(expectedAction);
  });

  it('should update the category selected', () => {
    const payload = 'technology';

    const expectedAction = {
      type: types.UPDATE_CATEGORY,
      payload,
    };

    expect(updateCategory(payload)).toEqual(expectedAction);
  });
});

describe('get category actions', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      clickedCategory: '',
      selectedIndex: 0,
    });
  });

  afterEach(() => {
    nock.cleanAll();
  });

  it('adds new articles to the category reducer', () => {
    nock(url)
      .persist()
      .get('/articles/categories/article?category=technology')
      .reply(200, { Categories: [{}] });
    return store.dispatch(getCategoryArticles('technology'))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});
