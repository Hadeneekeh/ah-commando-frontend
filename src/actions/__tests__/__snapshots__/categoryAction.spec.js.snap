// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get category actions adds new articles to the category reducer 1`] = `
Array [
  Object {
    "payload": Object {
      "Articles": undefined,
      "category": "technology",
    },
    "type": "UPDATE_CATEGORY_ARTICLES",
  },
]
`;
